catalog/nsili/catalog-nsili-endpoint/src/main/java/org/codice/alliance/nsili/endpoint/requests/OrderRequestImpl.java

```diff
@@ -495,92 +614,68 @@ protected void writeFile(FileLocation destination, InputStream fileData, long si
-
-            tarOut.flush();
-            fos.flush();
-            fos.close();
-            return fos.asByteSource();
         }
+
+        tarOut.flush();
     }
 
-    private ByteSource getGzip(InputStream data) throws IOException {
-        try (FileBackedOutputStream fos = new FileBackedOutputStream(MAX_MEMORY_SIZE);
-                GZIPOutputStream zipOut = new GZIPOutputStream(fos)) {
-            IOUtils.copy(data, zipOut);
-            zipOut.flush();
-            fos.flush();
-            return fos.asByteSource();
-        }
+    private void getTar(TarOutputStream tarOut, ResourceContainer file) throws IOException {
+        long modTime = System.currentTimeMillis() / 1000;
+        int permissions = DEFAULT_TAR_PERMISSION;
+
+        TarHeader fileHeader = TarHeader.createHeader(file.getName(),
+                file.getSize(),
+                modTime,
+                false,
+                permissions);
+        tarOut.putNextEntry(new TarEntry(fileHeader));
+        IOUtils.copy(file.getInputStream(), tarOut);
+
+        tarOut.flush();
+
     }
 
-    private ByteSource getZip(InputStream data, String name) throws IOException {
-        try (FileBackedOutputStream fos = new FileBackedOutputStream(MAX_MEMORY_SIZE);
-                ZipOutputStream zipOut = new ZipOutputStream(fos)) {
-            ZipEntry zipEntry = new ZipEntry(name);
-            zipOut.putNextEntry(zipEntry);
-            IOUtils.copy(data, zipOut);
-
-            zipOut.flush();
-            fos.flush();
-            fos.close();
-            return fos.asByteSource();
-        }
+    private void getGzip(GZIPOutputStream zipOut, InputStream data) throws IOException {
+        IOUtils.copy(data, zipOut);
+        zipOut.flush();
     }
 
-    private ByteSource getZip(List<ResourceContainer> files) throws IOException {
-        try (FileBackedOutputStream fos = new FileBackedOutputStream(MAX_MEMORY_SIZE);
-                ZipOutputStream zipOut = new ZipOutputStream(fos)) {
-            List<String> addedFiles = new ArrayList<>();
-            for (ResourceContainer file : files) {
-                if (!addedFiles.contains(file.getName())) {
-                    ZipEntry zipEntry = new ZipEntry(file.getName());
-                    zipOut.putNextEntry(zipEntry);
-                    IOUtils.copy(file.getInputStream(), zipOut);
-                    addedFiles.add(file.getName());
-                }
-            }
+    private void getZip(ZipOutputStream zipOut, InputStream data, String name) throws IOException {
 
-            zipOut.flush();
-            fos.flush();
-            fos.close();
-            return fos.asByteSource();
+        ZipEntry zipEntry = new ZipEntry(name);
+        zipOut.putNextEntry(zipEntry);
+        IOUtils.copy(data, zipOut);
+
+        zipOut.flush();
+
+    }
+
+    private void getZip(ZipOutputStream zipOut, List<ResourceContainer> files) throws IOException {
+
+        List<String> addedFiles = new ArrayList<>();
```

Comment:
Possibly, but seems to be out-of-scope for my task, since I was just changing from FBOS to TFBOS.